src/emqx_router.erl

```diff
@@ -247,10 +249,59 @@ delete_trie_route(Route = #route{topic = Topic}) ->
     end.
 
 %% @private
+-spec(maybe_trans(function(), list(any())) -> ok | {error, term()}).
+maybe_trans(Fun, Args) ->
+    case persistent_term:get(emqx_route_lock_type) of
```

Comment:
hot beam reload will not have the PT initialised (init code is in `emqx_router_sup`)
so we need a default value here.

Reply:
right, I should cherry pick from e4.2.6